azurerm/internal/services/web/resource_arm_function_app.go

```diff
@@ -78,7 +78,6 @@ func resourceArmFunctionApp() *schema.Resource {
 			"storage_connection_string": {
 				Type:      schema.TypeString,
 				Required:  true,
-				ForceNew:  true,
 				Sensitive: true,
 			},
```

Comment:
whilst this'd work it'd cause a bunch of runtime errors should someone opt to change the account being used - as such I'm wondering if we'd be better to split this field into two:

1) "storage_account_id" for the ID of the Storage Account in question - which is ForceNew
2) the "storage account access key" which is either the primary/secondary key and allows updating

WDYT?

Reply:
I love that idea! It makes more sense to me. Should I make this change with 2.0 in mind and remove the old property?

Reply:
now that we are post 2.0 what we will have to do is deprecate the old one and make them conflict.